services/digest_utils.py

```
@@ -0,0 +1,251 @@
+                    month = parts[2]
+                    day = parts[3]
+                    filename = parts[4]
+                    
+                    # Validate date components
+                    if not (year.isdigit() and month.isdigit() and day.isdigit()):
+                        logger.warning(f"Invalid date components in stories path: {video_path}")
+                        return ""
+                    
+                    # Construct intro PNG filename
+                    intro_png = f"{filename}_01_intro.png"
+                    return f"https://{self.worker_domain}/assets/stories/{year}/{month}/{day}/{intro_png}"
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)